unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/Skybox3DElements/SkyboxElementsReferences.cs

```diff
@@ -1,20 +1,27 @@
-﻿using UnityEngine;
+﻿using Cysharp.Threading.Tasks;
+using DCL.Providers;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using UnityEngine;
 
 namespace DCL.Skybox
 {
     public class SkyboxElementsReferences : MonoBehaviour
     {
-        private const string PREFAB = "SkyboxPrefabs/Skybox Elements";
+        private const string PREFAB = "SkyboxElements.prefab";
 
         public GameObject domeElementsGO;
         public GameObject satelliteElementsGO;
         public GameObject planarElementsGO;
 
-        public static SkyboxElementsReferences Create()
+        public static async Task<SkyboxElementsReferences> Create(IAddressableResourceProvider addressableResourceProvider)
         {
-            var refernces = Instantiate(Resources.Load<GameObject>(PREFAB)).GetComponent<SkyboxElementsReferences>();
-            refernces.gameObject.name = "_Skybox Elements";
+            CancellationTokenSource cts = new CancellationTokenSource();
+            cts.CancelAfterSlim(TimeSpan.FromSeconds(15));
```

Comment:
This returns an IDisposable, I think we have to Dipose it when done to prevent leaks. @mikhail-dcl can you confirm?

Reply:
TBH, the cancellation source was added for testing (there was a time this wasnt working, and I didnt know why so I added the timeout). It was left there by mistake, but I removed it.

It would be nice to have your opinion on it, @mikhail-dcl . Do you think its useful to leave it?

Reply:
`Timeout` is handled in the Addressables group settings: you don't need to handle it manually

Reply:
Done